src/cmd/ksh93/tests/vi.exp

```diff
@@ -0,0 +1,927 @@
+# Test for vi keybindings
+
+# vi mode requires being in control mode for ctrl-l to work
+# Send a [ctrl-L] to force the shell to redraw the prompt and command line. Doing so makes
+# it easier to verify the command line contains the expected text.
+proc redraw_prompt args {
+    global is_control_mode
+    if {$is_control_mode == 0} {
+        control_mode
+        send [ctrl L]
+        insert_mode
+    } else {
+        send [ctrl L]
+    }
+}
+
+# Enter vi control mode
+proc control_mode args {
+    global is_control_mode
+    send "\033"
+    set is_control_mode 1
+}
+
+# Enter vi insert mode
+proc insert_mode args {
+    global is_control_mode
+    send "i"
+    set is_control_mode 0
+}
+
+# Delete current line and change input mode to insert mode
+proc clear_prompt args {
+    send [ctrl C]
+    control_mode
```

Comment:
This causes problems, at least on macOS. What I'm seeing is that the [esc] is eaten, has no effect, which means the `i` sent by `insert_mode` is inserted in the command line. If I replace these two statements with `expect_prompt` the test reaches line 724 before failing on macOS. It fails on line 738 on Linux.

Reply:
Also, it should always be the case that a [ctrl-C] results in a new prompt that is already in insert mode. It should not be necessary to toggle between control and insert modes in this way. If doing this does make a difference that suggests a bug.